### ISSUE TITLE
Add "skip" and "only" functions on Deno.test

### ISSUE THREAD

**Issue by calebmchenry** [denoland/deno#5197]
Improve Deno testing experience with "skip" and "only" features

## skip

Adding "skip" to the test would cause the test to be skipped when the tests are run with `deno test`
```ts
Deno.test.skip('skipped test', () => {
  // ...
});
```

https://jestjs.io/docs/en/api.html#testskipname-fn
https://mochajs.org/#inclusive-tests

## only

Adding "only" to the test would cause only tests with "only" to be ran when running `deno test`
```ts
Deno.test.only('only test example', () => {
  // ...
});
```

https://jestjs.io/docs/en/api.html#testonlyname-fn-timeout
https://mochajs.org/#exclusive-tests

## info

```
deno --version
deno 1.0.0-rc1
v8 8.2.308
```

**Comment by nayeemrmn** [comment]
Ref https://github.com/denoland/deno/issues/4098#issuecomment-591019144. We have `skip`, and decided to add it as an option rather than this non-extensible JS nonsense with the function as a namespace containing similar functions. I've found myself needing something like `only` before, but that should also be an option.

**Comment by dsherret** [comment]
I'm a heavy user of stuff like `.only`. The key is to make it fast to add while working and quick to remove.

Having to refactor...

```ts
Deno.test("some message", () => {
});
```

...to...

```ts
Deno.test({
  only: true
  name: "some message",
  () => {},
});
```

...is impractical. I think a `Deno.test.only` would be best. because you just have to briefly add `.only` to the code.

**Comment by nayeemrmn** [comment]
One of the reasons I don't like the shorthand 😣 It's the same general problem of not being able to quickly add options to the shorthand, but more visible with `only`.

Since `Deno.test()` doesn't return anything currently, how about this:
```js
Deno.test("abc", () => {
  // ...
}).only();
```
Generally put, it can return a handle allowing you to edit options ergonomically (only for the shorthand?). This would be a much nicer definition.

**Comment by calebmchenry** [comment]
For anyone stumbling upon this issue in the future. Here is the docs for ignoring (skipping) tests https://github.com/denoland/deno/blob/master/docs/testing.md#ignoring-tests

I like the "ignoring tests based on options." I think it makes much more sense for tests that will be ignored longer than the time of writing the tests. I do think there are still use cases for `.ignore()` and `.only()`

## .only()
Isolating test while writing/fixing it. I agree with @dsherret 's comment https://github.com/denoland/deno/issues/5197#issuecomment-626264310

## ~.skip()~ .ignore()
Writing tests for features before writing the implementation of those features. This would require  ignoring all tests while each feature isn't implement. Then removing the ignores as you complete them.

**TLD;**
ignoring / only-ing via options is nice for long term ignores and only-s, but .ignore() and .only() are nice while developing tests. I am also not opposed to .only and .ignore being at the end as mentioned above.

**Comment by tylergoodman** [comment]
https://github.com/denoland/deno/issues/5197#issuecomment-626918637
https://github.com/denoland/deno/issues/5197#issuecomment-626264310

+1ing both of these. You're both very valid. It is very impractical to change the function signature so much. Kindly reconsider.

It seems workarounds are:
1. Just use the object literal signature everywhere and enjoy pasting `only: true,`
2. Some sort of helper that you'll need to import in all your test files. This seems to work:
```ts
// test_helper.ts
declare global {
  namespace Deno {
    namespace test {
      export function only(name: string, fn: Deno.TestDefinition['fn']): void;
    }
  }
}

Deno.test.only = function only(
  name: string,
  fn: Deno.TestDefinition['fn'],
): void {
  return Deno.test({
    fn,
    name,
    only: true,
  });
};
```

I don't like the "just use the `--filter` argument" answer. How am I supposed to change that on the fly, while `deno test --watch` is running? Is the expectation to stop watching, type in a minimal-length-but-sufficiently-matching filter string that matches my test name, and then re-run the test command?

There must be something I'm missing here, as surely you can see that simply appending `.only` here is way more convenient. Were there other arguments against adopting this common pattern?

**Comment by yacinehmito** [comment]
This issue has been marked as closed by #5793 but the PR didn't actually address the issue; see [this comment](https://github.com/denoland/deno/issues/5197#issuecomment-626264310).
